src/index.coffee

```diff
@@ -42,12 +42,19 @@ class ConnectApp
     @app.use connect.directory(if typeof @root == "object" then @root[0] else @root)
 
     if @https
-      @server = (http2 || https).createServer
-        key: @https.key || fs.readFileSync __dirname + '/certs/server.key'
-        cert: @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
-        ca: @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
-        passphrase: @https.passphrase || 'gulp'
-        @app
+
+      # first make sure it is always an object
+      if !(typeof (@https) is "object")
+        @https = {}
+
+      # use some defaults when not set. do not touch any other https options
+      # see https://github.com/AveVlad/gulp-connect/issues/172
+      @https.key        = @https.key || fs.readFileSync __dirname + '/certs/server.key'
+      @https.cert       = @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
+      @https.ca         = @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
+      @https.passphrase = @https.passphrase || 'gulp'
```

Comment:
I think this is a slight bug still, if someone specifies a key that doesn't have a passphrase it will always use gulp still. I would do one of two things:

If `@https` was just a bool, or some other truty object but the key is not set then use all the defaults.
```
if typeof (@https) is 'boolean' || !@https.key
  @https.key = fs.readFileSync __dirname + '/certs/server.key'
  @https.cert = fs.readFileSync __dirname + '/certs/server.crt'
  @https.ca = fs.readFileSync __dirname + '/certs/server.crt'
  @https.passphrase = 'gulp'
```

Or on the passphrase specifically, allow empty strings to be passed in:
```
@https.passphrase = @https.passphrase != null ? @https.passphrase : 'gulp'
```